### PR TITLE
Unify SQLQueryFactory signatures

### PR DESCRIPTION
**Pull request by timowest**

### CODE REVIEW

File: querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java [hunk 1]
```diff
@@ -90,4 +91,60 @@ public final Connection getConnection() {
...
+     */
+    public abstract AbstractSQLQuery<Integer, ?> selectZero();
+
+    /**
+     * Create a new SQLQuery instance with one as the projection
+     *
+     * @return select(1)
+     */
+    public abstract AbstractSQLQuery<Integer, ?> selectOne();
+    /**
```
Comment (Shredder121) [inline querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java]: What do you think of adding a space here?

**Comment by johnktims** [comment]
It looks like the subclasses are overriding the abstract methods in `AbstractSQLQueryFactory`'s with the same implementation. Maybe that functionality should be pulled up into the parent class.

**Comment by Shredder121** [comment]
They are overridden to specify the new return type.

**Comment by Shredder121** [comment]
I think this now looks good.

File: querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java [hunk 2]
```diff
@@ -90,4 +91,61 @@ public final Connection getConnection() {
         return connection.get();
     }
 
+    /**
+     * Create a new SQLQuery instance with the given projection
```
Comment (robertandrewbain) [inline querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java]: It's not returning a `SQLQuery` instance, it's returning a concrete impl of `AbstractSQLQuery` instance. Comment applies throughout changes to class.

File: querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java [hunk 3]
```diff
@@ -90,4 +91,61 @@ public final Connection getConnection() {
...
+     *
+     * @param expr projection
+     * @param <T>
```
Comment (robertandrewbain) [inline querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java]: Param is Expression<T>, not <T>

Comment (robertandrewbain) [inline querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java]: Incorrect param type

File: querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java [hunk 4]
```diff
@@ -90,4 +91,61 @@ public final Connection getConnection() {
...
+     * @return select(expr)
```
Comment (robertandrewbain) [inline querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java]: Elaborate

File: querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java [hunk 5]
```diff
@@ -90,4 +91,61 @@ public final Connection getConnection() {
...
+     *
+     * @return select(1)
+     */
+    public abstract AbstractSQLQuery<Integer, ?> selectOne();
+
+    /**
+     * Create a new SQLQuery instance with the given projection and source
+     *
+     * @param expr query source and projection
+     * @param <T>
```
Comment (robertandrewbain) [inline querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java]: ?

Comment (timowest) [inline querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java]: because the type parameter is also used in the return value

Comment (timowest) [inline querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQueryFactory.java]: AbstractSQLQuery has T for the projection type and Q for the concrete subclass type. Q is necessary for fluent method call invocations.

**Comment by johnktims** [comment]
This looks good now.

**Pull request merged**